### ISSUE TITLE
Remove obsolete sourceport files after an update

### ISSUE THREAD

**Issue by Shockwave508** [luxtorpeda-dev/luxtorpeda#356]
### Feature Request

Multiple versions of a sourceport could easily stack whenever a new update is installed, within the .cache folder, gradually draining onboard disk space over time.

### Solution

If a new version of a sourceport is downloaded through Luxtorpeda, delete the older, obsolete version after successful install.

**Comment by d10sfan** [comment]
Currently the client doesn't have a concept of versions, in that it does not see the older content in there.

I could do some other things like either provide a mode to delete the cache files as soon as they are downloaded and used, which would save on disk but add more network traffic. Or have some logic to remove inside the folders files that don't match.

What are your thoughts @Shockwave508 ?

**Comment by SkylarLK** [comment]
> Currently the client doesn't have a concept of versions, in that it does not see the older content in there.
> 
> I could do some other things like either provide a mode to delete the cache files as soon as they are downloaded and used, which would save on disk but add more network traffic. Or have some logic to remove inside the folders files that don't match.
> 
> What are your thoughts @Shockwave508 ?

I think the latter is a good stopgap measure.

**Comment by Shockwave508** [comment]
Latter is good. Will prevent obsolete copies piling up.

**Comment by SladeWilson52** [comment]
This has been open for nearly a year.

**Comment by d10sfan** [comment]
Yes that's correct, did you have something constructive to add here?

I'm still interested in doing this at some point but it's fairly complex since the current logic does not know about files that aren't in the json file and I've been pretty busy.

So I'll plan on getting to this when I can.

**Comment by d10sfan** [comment]
Also if someone is interested in looking into this in the meantime pull requests are always welcome

**Comment by SladeWilson52** [comment]
> Yes that's correct, did you have something constructive to add here?

My apologies, I got the impression that nothing was happening. I assume a rewrite of the logic will take a lot of time.
